mne/beamformer/tests/test_ap.py

```diff
@@ -0,0 +1,169 @@
+# Authors: Yuval Realpe <yuval.realpe@gmail.com>
+#
+# License: BSD-3-Clause
+
+import os.path as op
+
+import numpy as np
+from scipy import linalg
+from random import choice
+
+import mne
+from mne.beamformer import ap
+from mne.cov import regularize
+from mne.datasets import testing
+
+
+data_path = testing.data_path(download=False)
+fname_ave = op.join(data_path, 'MEG', 'sample', 'sample_audvis-ave.fif')
+fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis-cov.fif')
+fname_fwd = op.join(data_path, 'MEG', 'sample',
+                    'sample_audvis_trunc-meg-eeg-oct-4-fwd.fif')
+
+
+def _get_data(ch_decim=1):
+    """Read in data used in tests."""
+    # Read evoked
+    condition = choice(['Left Auditory',
+                        'Right Auditory',
+                        'Left visual',
+                        'Right visual'])
+    evoked = mne.read_evokeds(fname_ave, condition=condition,
+                              baseline=(None, 0))
+    evoked.crop(tmin=0.05, tmax=0.15)
+    picks = mne.pick_types(evoked.info, meg=True, eeg=False)
+    picks = picks[::ch_decim]
+    evoked.pick_channels([evoked.ch_names[pick] for pick in picks])
+    evoked.info.normalize_proj()
+
+    # Read noise_cov
+    noise_cov = mne.read_cov(fname_cov)
+    noise_cov['projs'] = []
+    noise_cov = regularize(noise_cov, evoked.info, rank='full', proj=False)
+
+    return evoked, noise_cov
+
+
+def simu_data_2src(evoked, forward, noise_cov, times, nave=1):
+    """Simulate an evoked dataset with 2 sources.
+
+    One source is put in each hemisphere.
+    """
+    # Generate the two dipoles data
+    mu, sigma = 0.1, 0.005
+    s1 = 1 / (sigma * np.sqrt(2 * np.pi)) * np.exp(-(times - mu) ** 2 /
+                                                   (2 * sigma ** 2))
+
+    mu, sigma = 0.075, 0.008
+    s2 = -1 / (sigma * np.sqrt(2 * np.pi)) * np.exp(-(times - mu) ** 2 /
+                                                    (2 * sigma ** 2))
+    data = np.array([s1, s2]) * 1e-9
+
+    src = forward['src']
+    rng = np.random.RandomState(42)
+
+    rndi = rng.randint(len(src[0]['vertno']))
+    lh_vertno = src[0]['vertno'][[rndi]]
+
+    rndi = rng.randint(len(src[1]['vertno']))
+    rh_vertno = src[1]['vertno'][[rndi]]
+
+    vertices = [lh_vertno, rh_vertno]
+    tmin, tstep = times.min(), 1 / evoked.info['sfreq']
+    stc = mne.SourceEstimate(data, vertices=vertices, tmin=tmin, tstep=tstep)
+
+    sim_evoked = mne.simulation.simulate_evoked(forward, stc, evoked.info,
+                                                noise_cov, nave=nave,
+                                                random_state=rng)
+
+    return sim_evoked, stc
+
+
+def _check_dipoles(dipoles, fwd, stc, evoked, residual=None):
```

Comment:
idem

Reply:
Added the 'rel_tol' parameter to the _check_dipoles() function defined in test_rap_music.py, then imported _check_dipoles() into test_ap.py, using it with different rel_tol then in test_rap_music.